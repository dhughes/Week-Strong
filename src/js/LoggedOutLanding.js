import React from 'react';
import styled from 'styled-components';
import logo from '../img/logo.svg';
import { Vbox } from './Box';
import Button from './Button';

const Container = styled(Vbox)`
  padding: 1em;
`;

const LogoContainer = styled(Vbox)`
  flex-grow: 1.5;
`;

const ButtonContainer = styled(Vbox)`
  flex-grow: 1;

  & > * {
    margin: 0.5rem auto;
  }
`;

const Landing = props => (
  <Container>
    <LogoContainer>
      <img src={logo} alt="Week-Strong logo" />
    </LogoContainer>
    <ButtonContainer>
      <Button default={true} onClick={props.onGetStartedClick}>Get Started Now</Button>
      <Button onClick={props.onLoginClick}>Login</Button>
    </ButtonContainer>
  </Container>
);

Landing.defaultProps = {
  onGetStartedClick: () => {},
  onLoginClick: () => {}
};

export default Landing;
