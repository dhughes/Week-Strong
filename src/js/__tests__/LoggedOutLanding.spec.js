import React from 'react';
import { shallow } from 'enzyme';
import LoggedOutLanding from '../LoggedOutLanding';

xtest('LoggedOutLanding renders', () => {
  const component = shallow(<LoggedOutLanding />);
  expect(component).toMatchSnapshot();
});

test('LoggedOutLanding shows logo', () => {
  const component = shallow(<LoggedOutLanding />);
  expect(component.find('img').length).toEqual(1);
});
