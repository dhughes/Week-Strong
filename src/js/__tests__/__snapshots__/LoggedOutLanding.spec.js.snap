// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutLanding renders 1`] = `
<div>
  Logged in Landing page
</div>
`;
